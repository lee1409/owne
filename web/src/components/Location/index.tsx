import { FC } from "react"
import { ButtonBase, ButtonBaseProps } from "@material-ui/core"
import { styled } from "@material-ui/core/styles"
import Icon from "../Icon"
import { LocationName } from "../Typography"

const StyledLocationName = styled(LocationName)(({ theme }) => ({
  marginLeft: theme.spacing(1),
}))

type LocationProps = {
  location: string
} & ButtonBaseProps

const Location: FC<LocationProps> = ({ ...props }) => {
  const { location } = props

  return (
    <ButtonBase {...props}>
      <Icon variant="Location" />
      <StyledLocationName>{location}</StyledLocationName>
    </ButtonBase>
  )
}

export default Location
