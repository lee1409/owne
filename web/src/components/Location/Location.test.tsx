import { render } from "@testing-library/react"
import Location from "."

test("Location Bar", () => {
  const location = render(<Location location="test" />)
  expect(location).toMatchSnapshot("Location")
})
