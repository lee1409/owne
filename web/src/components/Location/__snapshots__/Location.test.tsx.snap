// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location Bar: Location 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="MuiButtonBase-root"
        location="test"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root Icon-root-1 Icon-root-2"
          focusable="false"
          variant="Location"
          viewBox="0 0 24 24"
        >
          <span />
        </svg>
        <p
          class="MuiTypography-root WithStyles(ForwardRef(Typography))-root-5 WithStyles(ForwardRef(Typography))-root-6 Styled(WithStyles(ForwardRef(Typography)))-root-3 Styled(WithStyles(ForwardRef(Typography)))-root-4 MuiTypography-body1"
        >
          test
        </p>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="MuiButtonBase-root"
      location="test"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root Icon-root-1 Icon-root-2"
        focusable="false"
        variant="Location"
        viewBox="0 0 24 24"
      >
        <span />
      </svg>
      <p
        class="MuiTypography-root WithStyles(ForwardRef(Typography))-root-5 WithStyles(ForwardRef(Typography))-root-6 Styled(WithStyles(ForwardRef(Typography)))-root-3 Styled(WithStyles(ForwardRef(Typography)))-root-4 MuiTypography-body1"
      >
        test
      </p>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
